lib/core/init.js

```diff
@@ -1,57 +1,229 @@
+	});
+	RESUtils.init._queueModulesStage('beforeLoad', function(stage, module, moduleID) {
+		return RESUtils.deferred.collection([
+				RESUtils.init.await.options(moduleID),
+				RESUtils.init.await('documentHeadReady')
+			])
+			.then(RESUtils.init._runModuleStage.bind(this, stage, module, moduleID));
+	});
+	RESUtils.init._queueModulesStage('go', function(stage, module, moduleID) {
+		return RESUtils.deferred.collection([
+				RESUtils.init.await.beforeLoad(moduleID),
+				RESUtils.init.await('documentBodyReady')
+			])
+			.then(RESUtils.init._runModuleStage.bind(this, stage, module, moduleID));
+	});
+
+	RESUtils.init._queueModulesStage('afterLoad', function(stage, module, moduleID) {
+		return RESUtils.deferred.collection([
+				RESUtils.init.await.go(moduleID),
+				RESUtils.init.await('documentContentLoaded')
+			])
+			.then(RESUtils.init._runModuleStage.bind(this, stage, module, moduleID));
+	});
+
+	RESUtils.init.await._deferred('sourceLoaded').resolve(); // kick everything off
+};
 
+RESUtils.init._queueModulesStage = function(stage, queueModulesStage) {
+	var defs = $.map(modules, function(module, moduleID) {
+		var def = queueModulesStage(stage, module, moduleID);
+		RESUtils.init.await._deferred(stage, moduleID, def);
+		return def;
+	});
+	RESUtils.init.await._deferred(stage, '_all', RESUtils.deferred.collection(defs));
+};
+RESUtils.init._runModuleStage = function(stage, module, moduleID) {
+	var result, deferred;
+
+	if (typeof module[stage] === 'function') {
+		try {
+			result = module[stage]();
+		} catch (e) {
+			console.error('Error in modules[' + moduleID + '].' + stage, e, e.stack);
+			deferred = $.Deferred().reject(e);
+		}
+	}
+
+	if (typeof result === 'undefined') {
+		deferred = $.Deferred().resolve();
+	} else if (result.done) {
+		deferred = result;
+	} else if (result) {
+		deferred = $.Deferred().resolve(result);
+	} else {
+		deferred = $.Deferred().reject(result);
+	}
+
+	return deferred;
+}
+
+RESUtils.init._showErrors = function() {
+	var deferreds = RESUtils.init.await._deferreds,
+		deferred,
+		state,
+		failure,
+		failures = {};
+	for (var type in deferreds) {
+		if (!(deferreds.hasOwnProperty(type) && typeof deferreds[type] === 'object')) continue;
+		for (var id in deferreds[type]) {
+			if (!(deferreds[type].hasOwnProperty(id) && typeof deferreds[type][id].state === 'function')) continue;
+			if (failures[id]) continue;
+
+			deferred = deferreds[type][id];
+			state = deferred.state();
+			if (state !== 'resolved') {
+				failures[id] = type;
+				if (state === 'rejected') {
+					deferred.fail(function() {
+						failure = Array.prototype.slice.call(arguments, 0);
+						console.error('Failed to load', id, type, failure);
+					});
+				} else {
+					console.error('Failed to load', id, type, state);
+				}
+			}
+		}
+	}
+	RESUtils.init.await._deferreds = {}; // allow garbage collection
+}
+
+RESUtils.init.complete = function() {
 	// $.browser shim since jQuery removed it
 	$.browser = {
```

Comment:
It doesn't seem like any of our code uses `$.browser` so we should be able to drop this complete code without issues.